app.js

```diff
@@ -1,23 +1,170 @@
 var express = require('express'),
     app = express(),
+    system = process.env.BADGES_SYSTEM,
+    orcidRe = /(\d{4}-\d{4}-\d{4}-\d{4})@orcid\.org/,
     Client = require('badgekit-api-client');
 
 app.set('port', (process.env.PORT || 5000));
 app.use(express.static(__dirname + '/public'));
 
-var apiEndpoint = process.env.BADGES_ENDPOINT || 'http://localhost:8080';
+var apiEndpoint = process.env.BADGES_ENDPOINT || 'http://badgekit-api-sciencelab.herokuapp.com/';
 var auth = {
   key: process.env.BADGES_KEY || 'master',
-  secret: process.env.BADGES_SECRET || 'your secret'
+  secret: process.env.BADGES_SECRET || 'supersecret'
 };
 
 var client = new Client(apiEndpoint, auth);
 
+function emailFromORCID (orcid) {
+  return orcid + '@orcid.org';
+}
+
+function ORCIDFromEmail (email) {
+  var m;
+  if(m = orcidRe.exec(email) !== null){
+    return m[1]
+  }
+}
+
+function modEntry(entry, orcid){
+  entry.orcid = orcid;
+  delete entry.email;
+  return true;
+}
+
+function urlFromDOI(doi){
+  return "http://dx.doi.org/" + decodeURI(doi);
+}
+
+function DOIFromURL(url){
+  return encodeURI(url); // TODO: remove prefix doi.org/
+}
 
 app.get('/', function(request, response) {
   response.send('Welcome to PaperBadger!');
 });
 
+app.get('/badges', function(request, response){
+  client.getAllBadges({system: system}, function(err, badges){
+    if(err) console.error(err);
+    response.send(badges);
+  })
+});
+
+/* Get badges for a user */
+
+// Get all badge instances earned by a user
+app.get('/users/:orcid/badges', function(request, response){
+  var orcid = request.params.orcid;
```

Comment:
you want to check if orcid is not empty here before making the request. Early return and response with an error if that's the case.